File: python/leetcode/leet_0231.py
# Power of Two
# https://leetcode.com/problems/power-of-two/description/?envType=daily-question&envId=2024-02-19

def is_power_of_two(n: int) -> bool:
    if n <= 0:
        return False
    return (n & (n - 1)) == 0

File: rust/leet_code/src/leet_0231.rs
// Power of Two
// https://leetcode.com/problems/power-of-two/description/?envType=daily-question&envId=2024-02-19

fn is_power_of_two(n: i32) -> bool {
    if n <= 0 {
        return false;
    }

    (n & (n - 1)) == 0
}
